internal/service/cluster/resource_cluster.go

```diff
@@ -583,6 +599,16 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 		}
 	}
 
+	if pinnedFCVBlock, ok := d.Get("pinned_fcv").([]any); ok && len(pinnedFCVBlock) > 0 {
```

Comment:
[nit] looks like this isn't tested. However, since we reuse the `PinFCV` I am not that concerned.
The only minor issue I can see: A failed pinning will mark the cluster as tainted and require a replace

Reply:
good point. I am not too concerned about the create path because if a cluster is being created they would define the mongodb version they wish, dont see the use case of pinning fcv from the start.